src/Strategy/Elixir/MixExs.hs

```diff
@@ -0,0 +1,181 @@
+module Strategy.Elixir.MixExs (
+  analyze,
+  buildGraph,
+  mixExsFileParser,
+  MixExsPackage (..),
+  PackageName (..),
+  DepSCM (..),
+) where
+
+import Control.Effect.Diagnostics (Diagnostics, Has, context)
+import Data.Foldable (asum, for_)
+import Data.Map.Strict qualified as Map
+import Data.Maybe (listToMaybe)
+import Data.String.Conversion (toText)
+import Data.Text (Text)
+import Data.Void (Void)
+import DepTypes (DepEnvironment (..), DepType (GitType, HexType), Dependency (..), VerConstraint (..))
+import Effect.Logger (Logger, logDebug)
+import Effect.ReadFS (ReadFS, readContentsParser)
+import Graphing (Graphing, fromList)
+import Path
+import Prettyprinter (pretty)
+import Strategy.Elixir.ElixirParser (
+  ElixirAccessor (..),
+  ElixirAtom (..),
+  ElixirText (..),
+  ElixirValue (..),
+  betweenCurlyBrackets,
+  betweenSquareBrackets,
+  findKeyword,
+  findKeywordWithStringValue,
+  lexeme,
+  parseElixirValue,
+  symbol,
+ )
+import Strategy.Elixir.MixTree (DepSCM (..), PackageName (..), supportedSCM, toDependencyVersion)
+import Text.Megaparsec (
+  MonadParsec (takeWhileP, try),
+  Parsec,
+  anySingle,
+  sepEndBy,
+  skipManyTill,
+  (<|>),
+ )
+import Types (GraphBreadth (..))
+
+type Parser = Parsec Void Text
+
+-- | MixExsDeps represents dependency entry found in `mix.exs` file.
+data MixExsPackage = MixExsPackage
+  { mixExsPkgName :: PackageName
+  , mixExsPkgDepSCM :: DepSCM
+  , pkgExsVersionRequirement :: Maybe VerConstraint
+  , pkgExsOnlyEnvironment :: [DepEnvironment]
+  }
+  deriving (Show, Eq, Ord)
+
+mixExsFileParser :: Parser [MixExsPackage]
+mixExsFileParser = lexeme prefix *> deps <* symbol "end"
+  where
+    prefix = skipManyTill anySingle (symbol "defp deps do")
+    deps = betweenSquareBrackets (sepEndBy parseMixExsPackage (lexeme $ symbol ","))
+
+-- | Parses any unrecognized value as ElixirValue of type string.
+-- This is to avoid exception when parsing non relevant elixir type in deps entry. - e.g float, global variable, etc.
+unRecognizedElixirValue :: Parser ElixirValue
+unRecognizedElixirValue = (EString <$> ElixirText) . toText <$> takeWhileP (Just "ignored") (\c -> c /= ',' && c /= '}')
```

Comment:
Example would be: https://github.com/fossas/spectrometer/pull/322/files#diff-a7e0906db68780f4685851d93936ccec702719cdf0e230ce0b5f8b9938221e69R47

In current implementation, we parse all values, after which, we look for relevant options (e.g. associated git tag, etc.). So I think the risk is fairly _low_, but I can logs the value here. 

I can fail here instead.

Reply:
Interesting. I think not failing in this case is the right thing to do. I'm still a bit suspect about using "treat this as a string" to avoid failing as opposed to some other technique (e.g. maybe "throw away this expression") because I worry that treating unknown values as strings will cause a garbage parse, but this looks reasonable.